Automattic/newspack-plugin
# feat(registration-block): editable success state

### All Submissions:






### Changes proposed in this Pull Request:



Because this code introduces a `save` handler on the block, a warning will be shown in the editor. Since the relevant features (Reader Activation) are behind a flag and not officially released yet, I don't think this is an issue. 

Due to constraints of the block editor – a block can have only one block-editable area ("inner blocks") – there's only one success state. This means that the same message (with note about confirmation email) will be displayed regardless of whether the reader has an account or not. This is accurate, though, because the email will be sent anyway. Also, this way the response does not reveal if the email exists in the database. WDYT @miguelpeixe ? 

I'm not aware of a convention for editing block "states" – here's the switch implemented in this PR:

<img width="287" alt="image" src="https://user-images.githubusercontent.com/7383192/179253429-0aea72d7-9e3f-43a4-9d9f-c20ec65da4c8.png">

Please share your thoughts @thomasguillot 👆

Closes #1768.

### How to test the changes in this Pull Request:

1. On `master`, insert the Reader Registration block on a page
2. Switch to this branch, observe the block working as before on front-end 
3. Open the editor, observe a notice about needing block recovery – click the block recovery button and observe the recovery succeeding 
4. Save the post, observe it's still working as expected on front-end 
5. In the editor, observe a new sidebar panel – "Edited state" – with a "Initial"/"Success" toggle
6. Edit the success state by switching the edited state
7. On the front-end, observe the new success state is displayed after registering 

### Other information:
